.github/actions/test-package/action.yml

```diff
@@ -0,0 +1,76 @@
+name: test-spack
+description: 'Builds a spack package'
```

Comment:
Make more descriptive? e.g Current and past Spack builds

Once we're happy with this new setup we can remove the weekly Spack check in DOLFINx - edit: I guess that checks Spack upstream builtin. Maybe this can all me merged into here.